ddtrace/internal/utils/http.py

```diff
@@ -38,6 +39,15 @@ def strip_query_string(url):
     return h + fs + f
 
 
+def redact_url(url, query_string_obfuscation_pattern):
+    # type: (str, re.Pattern) -> str
+    hqs, fs, f = url.partition("#")
+    h, qss, qs = hqs.partition("?")
+
+    redacted_query = query_string_obfuscation_pattern.sub("<redacted>", compat.to_unicode(qs))
+    return h + qss + redacted_query + fs + f
```

Comment:
should we use urlparse tools for parsing and rejoining the url in a consistent way?

maybe when we have shared testing together it'll prove that this way is just fine ?

Reply:
done! I had to create a new function to adapt what we needed from urlparse, but it mostly is the same